### DISCUSSION TITLE
cursor doesn't move after accepting inline complete from copilot

### DISCUSSION THREAD

**Discussion by xuhdev** [emacs-lsp/lsp-mode#4656] [Q&A] [answered]
After copilot provides inline completion, hitting `<C-Enter>` accepts the inline completion as expected. However, the cursor remains at its current position and doesn't move to the end of the completion.

Is this expected?

**Comment by kiennq** [comment]
@kassick

**Reply by kassick** [reply, marked as answer]
It doesn't indeed (that was by design, as it matched my personal preference of "review what has just been inserted").

You can add this behavior by adding a hook to `lsp-inline-completion-accepted-hook`:

```elisp
(add-hook
 'lsp-inline-completion-accepted-hook 
 (lambda (inserted-text start end) (goto-char end)))
```

I may add a customize option to handle this

**Reply by kiennq** [reply]
I feel like moving to the end should be the default since it would allow the user to continue to type immediately after. The review of what's going to be inserted was happening before the acceptance happen via overlay already. With the mouse cursor not moving, we force the user to re-review it again and if it is multiple lines insert, there's no easy key to move to the end of the inserted text to continue working.
Also, in both cases, the user can still undo the insert via undo (is that the only way as well?).

**Reply by kiennq** [reply]
I put a proposed change in #4662
